target_chains/sui/sdk/js/src/client.ts

```diff
@@ -0,0 +1,298 @@
+   * @param vaas array of vaas to verify
+   * @param tx transaction block to add commands to
+   */
+  async verifyVaas(vaas: number[][], tx: TransactionBlock) {
+    const wormholePackageId = await this.getWormholePackageId();
+    const verifiedVaas = [];
+    for (const vaa of vaas) {
+      const [verifiedVaa] = tx.moveCall({
+        target: `${wormholePackageId}::vaa::parse_and_verify`,
+        arguments: [
+          tx.object(this.wormholeStateId),
+          tx.pure(vaa),
+          tx.object(SUI_CLOCK_OBJECT_ID),
+        ],
+      });
+      verifiedVaas.push(verifiedVaa);
+    }
+    return verifiedVaas;
+  }
+
+  /**
+   * Adds the necessary commands for updating the pyth price feeds to the transaction block.
+   * @param tx transaction block to add commands to
+   * @param updates array of price feed updates received from the price service
+   * @param feedIds array of feed ids to update (in hex format)
+   */
+  async updatePriceFeeds(
+    tx: TransactionBlock,
+    updates: number[][],
+    feedIds: HexString[]
+  ): Promise<ObjectId[]> {
+    const wormholePackageId = await this.getWormholePackageId();
+    const packageId = await this.getPythPackageId();
+
+    let priceUpdatesHotPotato;
+    if (updates.every((update) => this.isAccumulatorMsg(update))) {
+      if (updates.length > 1) {
+        throw new Error(
+          "SDK does not support sending multiple accumulator messages in a single transaction"
+        );
+      }
+      const vaa = this.extractVaaBytesFromAccumulatorMessage(updates[0]);
+      const verifiedVaas = await this.verifyVaas([vaa], tx);
+      [priceUpdatesHotPotato] = tx.moveCall({
+        target: `${packageId}::pyth::create_authenticated_price_infos_using_accumulator`,
+        arguments: [
+          tx.object(this.pythStateId),
+          tx.pure(updates[0]),
+          verifiedVaas[0],
+          tx.object(SUI_CLOCK_OBJECT_ID),
+        ],
+      });
+    } else if (updates.every((vaa) => !this.isAccumulatorMsg(vaa))) {
+      const verifiedVaas = await this.verifyVaas(updates, tx);
+      [priceUpdatesHotPotato] = tx.moveCall({
+        target: `${packageId}::pyth::create_price_infos_hot_potato`,
+        arguments: [
+          tx.object(this.pythStateId),
+          tx.makeMoveVec({
+            type: `${wormholePackageId}::vaa::VAA`,
+            objects: verifiedVaas,
+          }),
+          tx.object(SUI_CLOCK_OBJECT_ID),
+        ],
+      });
+    } else {
+      throw new Error("Can't mix accumulator and non-accumulator messages");
+    }
+
+    const priceInfoObjects: ObjectId[] = [];
+    for (const feedId of feedIds) {
+      const priceInfoObjectId = await this.getPriceFeedObjectId(feedId);
+      if (!priceInfoObjectId) {
+        throw new Error(
+          `Price feed ${feedId} not found, please create it first`
+        );
+      }
+      priceInfoObjects.push(priceInfoObjectId);
+      const coin = tx.splitCoins(tx.gas, [tx.pure(1)]);
```

Comment:
Is it assuming the update cost is always 1 per price feed? I think we should query it from on-chain.